src/backend/distributed/commands/extension.c

```diff
@@ -78,40 +76,801 @@ ErrorIfUnstableCreateOrAlterExtensionStmt(Node *parsetree)
+ * extension used by potentially multiple shards. To make sure all shards in
+ * the transaction can interact with the extension the extension needs to be
+ * visible on all connections used by the transaction, meaning we can only use
+ *  1 connection per node.
+ */
+static void
+EnsureSequentialModeForExtensionDDL(void)
+{
+	if (ParallelQueryExecutedInTransaction())
+	{
+		ereport(ERROR, (errmsg("cannot create extension because there was a "
+							   "parallel operation on a distributed table in the "
+							   "transaction"),
+						errdetail("When creating a distributed extension, Citus needs to "
+								  "perform all operations over a single connection per "
+								  "node to ensure consistency."),
+						errhint("Try re-running the transaction with "
+								"\"SET LOCAL citus.multi_shard_modify_mode TO "
+								"\'sequential\';\"")));
+	}
+
+	ereport(DEBUG1, (errmsg("switching to sequential query execution mode"),
+					 errdetail(
+						 "A distributed extension is created. To make sure subsequent "
+						 "commands see the type correctly we need to make sure to "
+						 "use only one connection for all future commands")));
+	SetLocalMultiShardModifyModeToSequential();
+}
+
+
+/*
+ * ShouldPropagateExtensionCommand determines whether to propagate an extension
+ * command to the worker nodes.
+ */
+static bool
+ShouldPropagateExtensionCommand(Node *parseTree)
+{
+	/* if we disabled object propagation, then we should not propagate anything. */
+	if (!EnableDependencyCreation)
+	{
+		return false;
+	}
+
+	/*
+	 * If extension command is run for/on citus, leave the rest to standard utility hook
+	 * by returning false.
+	 */
+	if (IsCreateAlterExtensionUpdateCitusStmt(parseTree))
+	{
+		return false;
+	}
+	else if (IsDropCitusStmt(parseTree))
+	{
+		return false;
+	}
+	else if (IsAlterExtensionSetSchemaCitus(parseTree))
+	{
+		return false;
+	}
+
+	return true;
+}
+
+
+/*
+ * IsCreateAlterExtensionUpdateCitusStmt returns whether a given utility is a
+ * CREATE or ALTER EXTENSION UPDATE statement which references the citus extension.
+ * This function returns false for all other inputs.
+ */
+bool
+IsCreateAlterExtensionUpdateCitusStmt(Node *parseTree)
+{
+	const char *extensionName = NULL;
+
+	if (IsA(parseTree, CreateExtensionStmt))
+	{
+		extensionName = ((CreateExtensionStmt *) parseTree)->extname;
+	}
+	else if (IsA(parseTree, AlterExtensionStmt))
 	{
-		DefElem *defElement = (DefElem *) lfirst(optionsCell);
-		if (strncmp(defElement->defname, "new_version", NAMEDATALEN) == 0)
+		extensionName = ((AlterExtensionStmt *) parseTree)->extname;
+	}
+	else
+	{
+		/*
+		 * If it is not a Create Extension or a Alter Extension stmt,
+		 * it does not matter if the it is about citus
+		 */
+		return false;
+	}
+
+	/*
+	 * Now that we have CreateExtensionStmt or AlterExtensionStmt,
+	 * check if it is run for/on citus
+	 */
+	return (strncmp(extensionName, CITUS_APPLICATION_NAME, NAMEDATALEN) == 0);
```

Comment:
Citus application name might change in the future, let's use
```C
strncasecmp(extensionName, CITUS_NAME, NAMEDATALEN) == 0
```

Reply:
But CITUS_NAME is equal to "Citus", if I am not missing something ?

Reply:
I just aware that you suggested me to use `strncasecmp` for this reason, sorry for this